src/libtorch.cc

```diff
@@ -1284,62 +1621,114 @@ ModelInstanceState::ReadOutputTensors(
+      TRITONSERVER_DataType config_datatype = output_dtype_map_[name];
+      if (config_datatype != output_dtype) {
+        RETURN_IF_ERROR(TRITONSERVER_ErrorNew(
+            TRITONSERVER_ERROR_INVALID_ARG,
+            (std::string("configuration expects datatype TYPE_") +
+             TRITONSERVER_DataTypeString(config_datatype) + " for output '" +
+             name + "', model provides TYPE_" +
+             TRITONSERVER_DataTypeString(output_dtype))
+                .c_str()));
+      }
 
-    const char* output_buffer =
-        static_cast<const char*>(output_flat.data_ptr());
+      const char* output_buffer =
+          static_cast<const char*>(output_flat.data_ptr());
 
-    // Output tensors may not reside on the same device as model
-    torch::Device tensor_device = output_flat.device();
+      // Output tensors may not reside on the same device as model
+      torch::Device tensor_device = output_flat.device();
 
-    //  Set output shape
-    std::vector<int64_t> batchn_shape;
-    auto shape = output_tensors[op_index].sizes();
-    for (auto itr = shape.begin(); itr != shape.end(); itr++) {
-      batchn_shape.push_back(*itr);
-    }
+      // Get output shape
+      std::vector<int64_t> batchn_shape;
+      auto shape = output_tensors[op_index].toTensor().sizes();
+      for (auto itr = shape.begin(); itr != shape.end(); itr++) {
+        batchn_shape.push_back(*itr);
+      }
+
+      if (batchn_shape.size() == 0) {
+        return TRITONSERVER_ErrorNew(
+            TRITONSERVER_ERROR_INVALID_ARG,
+            (std::string("output '") + name +
+             "' is a scalar which is not supported.")
+                .c_str());
+      }
+
+      responder.ProcessTensor(
+          name, output_dtype, batchn_shape, output_buffer,
+          (tensor_device.type() == torch::kCPU) ? TRITONSERVER_MEMORY_CPU
+                                                : TRITONSERVER_MEMORY_GPU,
+          (tensor_device.type() == torch::kCPU) ? 0 : tensor_device.index());
+
+    } else if (output_tensors[op_index].isList()) {
+      // Custom handling for string/bytes tensor...
+
+      torch::List<torch::jit::IValue> output_list =
+          output_tensors[op_index].toList();
+
+      // Get output shape
+      std::vector<int64_t> batchn_shape{(int64_t)output_list.size()};
+      if (output_list.elementType()->kind() != c10::TypeKind::StringType) {
+        return TRITONSERVER_ErrorNew(
+            TRITONSERVER_ERROR_INVALID_ARG,
+            (std::string("output '") + name +
+             "' must be of type Tensor or List[str], recieved List[" +
+             output_list.elementType()->str() + "]")
+                .c_str());
+      }
 
-    if (batchn_shape.size() == 0) {
-      RETURN_IF_ERROR(TRITONSERVER_ErrorNew(
+      size_t tensor_offset = 0;
+
+      for (size_t idx = 0; idx < responses->size(); idx++) {
+        auto& response = (*responses)[idx];
+
+        const size_t tensor_element_cnt = GetElementCount(batchn_shape);
```

Comment:
Shouldn't use element count for the whole batch, again, batching can be enable even for 1-D tensor, you can have a batch of [2], [1] and the output shape should be set accordingly instead of 3.

Reply:
The batchn_shape is derived from the actual output of the pytorch model. It would always be the right shape.

Reply:
See [comment](https://github.com/triton-inference-server/pytorch_backend/pull/60/files#r863965574) that the model can allow batching even the input is 1-D string, then in my above example (assume identity), the model returns [3] for `batchn_shape`, I don't think here properly scatter the output to different responses.

Reply:
The actual scatter happens here: https://github.com/triton-inference-server/tensorflow_backend/blob/main/src/tensorflow.cc#L2059
We are anyways setting a check against dynamic batching while loading the model. I think we must not concern too much right now with DB if the user doesn't want it. We can add the feature later on?

Reply:
I agree. Since we are close to code freeze and these changes are stable as is. I will file a ticket for dynamic batching and the reshaped + batching use case that @GuanLuo pointed out.